### PR TITLE
Added testcases for config and also added related view functions in c…

### PR DESCRIPTION
**Pull request by siddhpurakaran**
# Changes

List of Changes:
 - Added testcases for Config contract so now test coverage for config contract is 100%
 - Fixed  updateFee and updateAmmDepositThreshold functions (instead of memory updated to storage keyword)
 - Added required getter and setter functions (earlyRedemptionFee, updateRepurchaseFeePercentage)
 - Added Prettier config

### CODE REVIEW

File: test/contracts/CorkConfig.ts [hunk 1]
```diff
@@ -3,28 +3,62 @@ import {
...
 import hre from "hardhat";
+import { parseEther, parseUnits, zeroAddress } from "viem";
 import * as helper from "../helper/TestHelper";
 
 describe("CorkConfig", function () {
   let defaultSigner: any;
   let secondSigner: any;
   let signers: any;
 
+  let expiryTime: any;
```
Comment (ziankork) [inline test/contracts/CorkConfig.ts:14]: please add types to this, like the new uni v2 integration

**Review by ziankork — CHANGES_REQUESTED** [review]

File: test/contracts/CorkConfig.ts (hunk 1 above)
Comment (ziankork) [reply test/contracts/CorkConfig.ts:14]: The way i would recommend this is by using Utility types from typescript, couple of it you might find useful is Awaited and ReturnType. please read [this](https://www.typescriptlang.org/docs/handbook/utility-types.html).

Comment (siddhpurakaran) [reply test/contracts/CorkConfig.ts]: Done : Added types for variables in CorkConfig.ts file and and types are already added in other test files through uniswap v2 PR so just focused on types in config test file

Comment (ziankork) [reply test/contracts/CorkConfig.ts]: Nice, please resolve the conflicts below. i recommend pulling from main first on your IDE and resolve the conflicts there. Don't resolve the conflicts through the Github built-in editor

Comment (siddhpurakaran) [reply test/contracts/CorkConfig.ts]: Fixed merge conflicts. Thanks

Comment (ziankork) [reply test/contracts/CorkConfig.ts]: There's another merge conflict caused by the freeze method PR, please fix them

Comment (siddhpurakaran) [reply test/contracts/CorkConfig.ts]: Merge conflict resolved

**Pull request merged**